x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotRetentionConfiguration.java

```diff
@@ -114,32 +116,48 @@ public Integer getMaximumSnapshotCount() {
      */
     public Predicate<SnapshotInfo> getSnapshotDeletionPredicate(final List<SnapshotInfo> allSnapshots) {
         final int snapCount = allSnapshots.size();
-        List<SnapshotInfo> sortedSnapshots = allSnapshots.stream()
+        final List<SnapshotInfo> sortedSnapshots = allSnapshots.stream()
             .sorted(Comparator.comparingLong(SnapshotInfo::startTime))
             .collect(Collectors.toList());
+        final long nonFailedSnapshotCount = allSnapshots.stream()
```

Comment:
This should be named "nonSuccessulSnapshotCount" since it doesn't include PARTIAL snapshots I think?

Reply:
Renamed this to `successfulSnapshotCount`.